modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -69,16 +77,16 @@ extension AddressBookClient: DependencyKey {
                         let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                         
                         if let contactsData = try? Data(contentsOf: unencryptedFileURL) {
-                            // file exists, ensure data are parsed, re-saved and file deteled
-                            let contacts = try AddressBookClient.contactsFrom(unencryptedData: contactsData)
+                            // Unencrypted file exists; ensure data are parsed, re-saved as encrypted, and file deteled
+                            let contacts = try AddressBookClient.contactsFrom(contactsData)
 
                             // try to encrypt and store the data
                             do {
                                 try AddressBookClient.storeContacts(contacts, remoteStorage: remoteStorage, remoteStore: false)
```

Comment:
Why is this not also trying to store remotely? I don't understand why the `remoteStore` flag needs to exist. It's easier to reason about if we always try to do a remote store whenever we do a local store.

Reply:
Actually, we need to sync contacts and re-store instead of store, that would replace remote file with local one. We need merge strategy to be applied. Code updated.

Reply:
Thanks for catching my error.